src/main/java/com/mongodb/hadoop/output/MongoOutputCommitter.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright 2011-2013 10gen Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hadoop.output;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.BulkUpdateRequestBuilder;
+import com.mongodb.BulkWriteOperation;
+import com.mongodb.BulkWriteRequestBuilder;
+import com.mongodb.DBCollection;
+import com.mongodb.DBObject;
+import com.mongodb.MongoException;
+import com.mongodb.hadoop.io.BSONWritable;
+import com.mongodb.hadoop.io.MongoUpdateWritable;
+import com.mongodb.hadoop.io.MongoWritableTypes;
+import com.mongodb.hadoop.util.MongoConfigUtil;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.OutputCommitter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+
+import java.io.IOException;
+
+/**
+ * MongoOutputCommitter to committ output
+*/
+public class MongoOutputCommitter extends OutputCommitter {
+
+    public static final String TEMP_DIR_NAME = "_MONGO_OUT_TEMP";
+    private static final Log LOG = LogFactory.getLog(MongoOutputCommitter.class);
+    private DBCollection collection;
+
+    public MongoOutputCommitter() {}
+
+    @Override
+    public void setupJob(final JobContext jobContext) {
+        LOG.info("Setting up job.");
+    }
+
+    @Override
+    public void setupTask(final TaskAttemptContext taskContext) {
+        LOG.info("Setting up task.");
+    }
+
+    public boolean needsTaskCommit(
```

Comment:
All overrides with use if compatUtil have been removed e.g., https://github.com/mongodb/mongo-hadoop/blob/20208a027ad8638e56dfcf040773f176d6ee059f/core/src/main/java/com/mongodb/hadoop/output/MongoOutputCommitter.java#L61

Reply:
minor nit: Better to keep `@Override` annotations for these methods - `needTaskCommit`, `commitTask`, `abortTask` since they are from base class `OutputCommiter`

Reply:
You mean to keep `@Override` on top of the method declaration?

Reply:
yes thats correct.